core/src/main/kotlin/org/evomaster/core/output/TestSuiteOrganizer.kt

```diff
@@ -111,46 +159,81 @@ class SortingHelper {
      * coveredTargets sorts [EvaluatedIndividual] objects on the basis of the number of covered targets.
      * The purpose is to give an example of sorting based on fitness information.
      */
-    val coveredTargets = compareBy<EvaluatedIndividual<*>> {
+    private val coveredTargets: Comparator<EvaluatedIndividual<*>> = compareBy {
         it.fitness.coveredTargets()
     }
 
     /**
      *  [comparatorList] holds those comparators that are currently selected for sorting
-     *  Note that (currently) the order of the comparators is inverse to their importance/priority
+     *  Note that the order of the comparators is the order their importance/priority.
      */
 
-    val comparatorList = mutableListOf(  maxStatusCodeComparatorInd, dbInitSize)
+    //var comparatorList = mutableListOf(statusCode, minActions)
+    var comparatorList = mutableListOf(statusCode, coveredTargets)
+
+    private val availableSortCriteria = mutableListOf(statusCode, minActions, maxStatusCode, maxActions, dbInitSize, coveredTargets)
 
+    fun getAvailableCriteria(): MutableList<Comparator<EvaluatedIndividual<*>>> {
+        return availableSortCriteria
+    }
+
+    fun selectCriteria(selected: MutableList<Comparator<EvaluatedIndividual<*>>>){
+        if (availableSortCriteria.containsAll(selected)){
+            comparatorList = selected
+        }
+        else {
+            throw UnsupportedOperationException("The sorting criteria chosen appear to not be supported at the moment.")
+        }
+    }
+
+    fun selectCriteriaByIndex(selected: MutableList<Int>){
+        if (availableSortCriteria.indices.toMutableList().containsAll(selected)){
+            comparatorList = availableSortCriteria.filterIndexed{ index, _ ->
+                selected.contains(index)
+            } as MutableList<Comparator<EvaluatedIndividual<*>>>
+        }
+        else {
+            throw UnsupportedOperationException("The sorting criteria chosen appear to not be supported at the moment.")
+        }
+    }
 
     /**
      *Sorting is done according to the comparator list. If no list is provided, individuals are sorted by max status.
      */
-    fun sortByComparatorList (solution: Solution<*>,
+    private fun sortByComparatorList (solution: Solution<*>,
                               namingHelper: NamingHelper,
-                              comparators: MutableList<Comparator<EvaluatedIndividual<*>>> = mutableListOf(maxStatusCodeComparatorInd)
+                              comparators: MutableList<Comparator<EvaluatedIndividual<*>>> = mutableListOf(statusCode)
 
     ): List<TestCase> {
         var counter = 0
 
-        comparators.forEach { solution.individuals.sortWith(it) }
+        val inds = solution.individuals
+
+        comparators.asReversed().forEach {
```

Comment:
bit confused... wouldn't the last sorting override/change all the previous sortings? if not, add a comment to clarify it

Reply:
ok. just make sure to have a test case checking this behavior